src/com/xilinx/rapidwright/rwroute/PartialRouter.java

```diff
@@ -78,6 +85,44 @@ protected void addStaticNetRoutingTargets(Net staticNet){
 		}
 	}
 	
+	private List<SitePinInst> findUnroutedSinks(Net net, List<SitePinInst> sinks) {
+	    Set<Node> usedNodes = new HashSet<>();
+	    List<SitePinInst> unroutedSinks = new ArrayList<>();
+	    for(PIP p : net.getPIPs()) {
+	        usedNodes.add(p.getStartNode());
+	        usedNodes.add(p.getEndNode());
+	    }
+	    for(SitePinInst sink : sinks) {
+	        if(!usedNodes.contains(sink.getConnectedNode())) {
+	            unroutedSinks.add(sink);
+	        }else {
+	            sink.setRouted(true);
+	        }
+	    }
+	    return unroutedSinks;
+	}
+
+	@Override
+	protected void routeStaticNets() {
+		if (staticNetAndRoutingTargets.isEmpty())
+			return;
+
+		Net gnd = design.getGndNet();
+		Net vcc = design.getVccNet();
+
+		// Copy existing PIPs
+		List<PIP> gndPips = (staticNetAndRoutingTargets.containsKey(gnd)) ? new ArrayList<>(gnd.getPIPs()) : Collections.emptyList();
+		List<PIP> vccPips = (staticNetAndRoutingTargets.containsKey(vcc)) ? new ArrayList<>(vcc.getPIPs()) : Collections.emptyList();
```

Comment:
Perhaps we should move all PIPs out (replacing the existing list with an empty one) rather than copying them?

Reply:
> Perhaps we should move all PIPs out (replacing the existing list with an empty one) rather than copying them?
Sounds like a good idea to me 👍 .